app/controllers/concerns/pagination_utility.rb

```diff
@@ -0,0 +1,67 @@
+module PaginationUtility
+  extend ActiveSupport::Concern
+
+  # This constant is used to specify "all" items to be shown in a view that
+  # uses the will_paginate gem for listing a collection.  That has to be a number.
+  # This particular number is used when the user selects "All" in the
+  # items-per-page selection.
+  ALL_ITEMS = 10_000.freeze
+
+  def process_pagination_params(entity)
+
+    # This method is used in controller actions involved in pagination of
+    # collection tables (e.g., companies, member_applications).
+
+    # It is passed a string that indicates the type of paginated collection,
+    #  e.g. "company", "membership_application".
+
+    # It returns:
+    #  1) search params hash, for use with ransack gems "ransack" method,
+    #  2) the user's last items-per-page selection (an integer or 'All'),
+    #  3) the actual number of items-per-page to show in the table.
+
+    # This method uses the session to store (and recover) search criteria
+    # and per-page items selection.  These need to be persisted across action
+    # invocations in order to accommodate the multiple ways in which a
+    # typical controller "index" action might be called.
+
+    # For instance, the companies_controller "index" action is called when:
+    # 1) loading the index page,
+    # 2) moving to another pagination page in the companies listing table (XHR),
+    # 3) sorting on one of the table columns,
+    # 4) executing a companies search from the index page, and,
+    # 5) changing per-page items count in the pagination table on that page (XHR).
+
+    entity_items_selection = (entity + '_items_selection').to_sym
+    entity_search_criteria = (entity + '_search_criteria').to_sym
+
+    if params[:items_count]  # << user has selected a per-page items count
+      items_count = params[:items_count]
+      items_selection = items_count == 'All' ? 'All' : items_count.to_i
+
+      session[entity_items_selection] = items_selection
+
+      search_criteria = JSON.parse(session[entity_search_criteria])
+
+      search_params = search_criteria ?
+        ActionController::Parameters.new(search_criteria) : nil
+
+      # Reset params hash so that sort_link works correctly in the view
+      # (the sort links are built using, as one input, the controller params)
+      params[:q] = search_params
+      params.delete(:items_count)
+
+    else
+      items_selection = session[entity_items_selection] ?
+        session[entity_items_selection] : 10
```

Comment:
I wasn't going to since this is the only place it is referenced.  However, adding a constant provides a chance to indicate what the number represents (as was probably your point).